### PR TITLE
Update query cli commands to handle Allegra and Mary eras

### PR DESCRIPTION
**Pull request by Jimbo4350**

### CODE REVIEW

File: cardano-api/src/Cardano/Api/CliMode.hs [hunk 1]
```diff
@@ -27,9 +27,9 @@ import           Cardano.Api.Typed
 import           Ouroboros.Consensus.Node.Run (RunNode)
 
 
-data Protocol = ByronProtocol !EpochSlots
-              | ShelleyProtocol
-              | CardanoProtocol !EpochSlots
+data CliMode = CliByronMode !EpochSlots
+             | CliShelleyMode
+             | CliCardanoMode !EpochSlots
```
Comment (intricate) [inline cardano-api/src/Cardano/Api/CliMode.hs:32]: I don't really agree with this name change. `Protocol` still seems appropriate imo.

Comment (dcoutts) [reply cardano-api/src/Cardano/Api/CliMode.hs:32]: I agree. Lets not change the names in this PR.

I also have a good solution that avoids having to come up with new names, and will not clash with the existing `NodeConsensusMode` type. But we can do that later.

**Review by dcoutts — APPROVED** [review]
Tricky one.

**Comment by dcoutts** [comment]
bors merge

**Pull request merged**